launch_testing/launch_testing/loader.py

```diff
@@ -285,6 +292,17 @@ def _iterate_test_classes_in_test_suite(test_suite):
             yield t.__class__
 
 
+def _iterate_test_suites(test_suite):
+    try:
+        iter(test_suite)
+    except TypeError:
```

Comment:
what is this catching? Shouldn't `pre_shutdown_tests`/`post_shutdown_tests` always be an iterable?

Reply:
The `test_suite` object is not iterable if it is actually a test instance and not a test suite containing a test instances.

This is the same logic that is used in the enumeration of only the test suites (the opposite of this new function): https://github.com/ros2/launch/blob/d16628d8abfcdd7d44293c89198a1a64cfc54ee6/launch_testing/launch_testing/loader.py#L288-L293

Reply:
Ah, the CI has demonstrated that the inverse of the logic in `_iterate_tests_in_test_suite` doesn't hold true here. I'll have to find a better way to enumerate the suites.

Reply:
Resolved by b604a34